src/microservices/Microservices.IdentifierMapper/Execution/Swappers/TableLookupWithGuidFallbackSwapper.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+using FAnsi.Discovery;
+using Smi.Common.Options;
+
+namespace Microservices.IdentifierMapper.Execution.Swappers
+{
+    /// <summary>
+    /// Identifier swapper which connects to lookup table (wrapping <see cref="TableLookupSwapper"/>) but if no mapping is
+    /// found then generates a guid substitution value (which is stored in the same database - using <see cref="ForGuidIdentifierSwapper"/>)
+    ///
+    /// <para>The Guid mapping table will be the mapping table name with the suffix <see cref="GuidTableSuffix"/> (i.e. <see cref="IMappingTableOptions.MappingTableName"/> + suffix)</para>
+    /// </summary>
+    public class TableLookupWithGuidFallbackSwapper : ISwapIdentifiers
```

Comment:
I think this needs to support the GUID table not being on the same database server as the mapping table. For our use-case at least, we have read-only access to the mapping table

Reply:
Resolved after catch-up call